src/forms/common/DebounceUtils.tsx

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+const defaultDebounce = 500; // miliseconds
+
+export class DebounceUtils {
+    debounceTimer: any = null;
+
+    /**
+     * This method only suports the debounce with Events
+     * @param evt - SyntheticEvent Event which is input to the called function
+     * @param func - Function to be applied post debouce time is over
+     * @param debounceTime - optional debounceTime in miliseconds, defaultDebounce will be used if not provided
+     */
+    public debounce = (evt: React.SyntheticEvent, func: Function | undefined, debounceTime: number | undefined) => {
+        // if function passed is not set by parent
+        if (!func) {
+            return func;
+        }
+        const waitTime = debounceTime || defaultDebounce;
```

Comment:
If debounceTime=0 is passed then defaultDebounce=500 will be set. Is this by design ?

Reply:
if debounce 0 is passed, it will use 0 and setTimeout triggers immediately when i get chance. 
500 is applied only when the debounceTime is not set at all.  Its safe to not set debounceTime at all and let component set 500ms by default.  

User need to set this when they want longer or lesser debounce time then default.